spade/core/project.py

```diff
@@ -111,3 +77,65 @@ def get_template_for_file(self, f, template):
         Gets the associated template for a file.
         """
         pass
+
+    def __add_info(self, key, value, nomodify=False):
+        # TODO(nyxxxie): handle nomodify var
+        ins = self.table_pinfo.insert()
+        conn = self.__db_engine.connect()
+        conn.execute(ins, key=key, value=value) # TODO: might cause problems?
+
+    def __init_db(self, engine):
+        """
+        Creates default tables for a newly created spade project
+        """
+        file_hash_len = 256/8
+        change_hash_len = 256/8
+        metadata = MetaData()
+        # Define project_info table
+        self.table_pinfo = Table("project_info", metadata,
+            Column("key", String, primary_key=True),
+            Column("value", String)
+        )
+        # Define files table
+        self.table_files = Table("files", metadata,
+            Column("id", Integer, primary_key=True, autoincrement=True),
+            # Relative or absolute path to this file
+            Column("path", String, unique=True),
+            # sha256 hash of file contents on last change
+            Column("hash", Binary),
+            # head change file is at
+            Column("head_change", Binary, server_default=None)
+        )
+        # Define changes table
+        self.table_changes = Table("changes", metadata,
+            # id of file we apply changes to
+            Column("file_id", None, ForeignKey("files.id")),
+            # sha256 of this change
+            Column("hash", Binary),
+            # sha256 of previous change
+            Column("parent", Binary),
+            # position in file where change occured
+            Column("file_pos", Integer),
+            # NOTE(fst3a): Already solved this
+            # change type.  '+' = insert, '-' = erase, '!' = replace
+            Column("change_type", Integer),
+            # bytes that were inserted or erased
+            Column("change", Binary)
+        )
+        # FIXME(fst3a): This is plain wrong, one project can have multiple files
```

Comment:
If you're undecided on how this feature should function, perhaps we should just make it a seperate feature and merge it when it is ready?  I don't like large amounts of commented out code accompanied by a large FIXME comment just sitting in code that might be merged soon.